app/config/auth.js

```diff
@@ -0,0 +1,55 @@
+const Joi = require('joi')
+const { getEnvironmentVariable } = require('../lib/environment-helpers')
+
+const jsonArray = Joi.extend({
+  type: 'array',
+  base: Joi.array(),
+  coerce: {
+    from: 'string',
+    method (value) {
+      if (
+        typeof value !== 'string' ||
+        (value[0] !== '[' && !/^\s*\[/.test(value))
+      ) {
+        return
+      }
+
+      try {
+        return { value: JSON.parse(value) }
+      } catch (ignoreErr) { }
+    }
+  }
+})
+
+// Define config schema
+const schema = Joi.object({
+  authTokens: Joi.object({
+    portalKey: Joi.string().required(),
+    enforcementKey: Joi.string().required()
+  }),
+  permittedDomains: jsonArray.array().items(Joi.string().pattern(/^[a-z0-9.@]+([-.][a-z0-9]+)*\.[a-z]{2,6}$/)).default([])
```

Comment:
Nice - I see you have a test for this regex?

Reply:
am testing it in test/unit/config/auth.test.js

Reply:
should fail if PERMITTED_DOMAINS has a value which is not a domain

Reply:
Haha - sorry - didn't mean to add a question mark. I wasn't asking, I was stating :)